src/mqtt_client.rs

```diff
@@ -546,6 +548,10 @@ impl<
 }
 
 /// The general structure for managing MQTT via Minimq.
+///
+/// # Note
+/// To connect and maintain an MQTT connection, the `Minimq::poll()` method must be repeatedly
+/// called.
```

Comment:
```suggestion
/// To connect and maintain an MQTT connection, the `Minimq::poll()` method must be
/// called regularly. It should also be called repeatedly until it returns `Ok(None)` (or `Err()`).
```

Reply:
I don't think that update is accurate - `Ok(None)` will also be returned when processing a control packet (e.g. CONNACK), but doesn't necessarily mean there's no packets left.

Reply:
Ack. But then we should make the comment true by improving the code.
Can we make `poll()` process packets until `client.handle_packet()` returns `Ok(Some())`, i.e. until it did call `f()`?

Reply:
I've updated the docs on `poll()` and rephrased this to indicate that `poll()` must be called regularly.

Reply:
That's not a bad idea, but I'd say it's out of scope here. Will open an issue for it.

Reply:
I spawned https://github.com/quartiq/minimq/issues/102 to handle this